File: Array/leetcode/medium/MajorityElements.py
#https://leetcode.com/problems/majority-element/description/
def majority_element(nums):
    # Implementation
    count = 0
    candidate = 0
    for item in nums:
        if count == 0:
           candidate = item
        elif candidate == item:
            count += 1
        else:
            count -= 1
    return candidate

test_cases = [
    [3, 2, 3],
    [2, 2, 1, 1, 1, 2, 2],
    [1],
    [6, 5, 5],
]

for i, nums in enumerate(test_cases, 1):
    result = majority_element(nums)
    print(f"Test Case {i}: {result}")

File: Array/leetcode/medium/MajorityElements.go
// https://leetcode.com/problems/majority-element/description/
package main

import "fmt"

func majorityElement(nums []int) int {
     count := 0
	 candidate := 0

	 for _, value := range(nums) {
		 if count == 0 {
			 candidate = value
		 }
		 if value == candidate {
			 count++
		 } else {
			 count--
		 }
	 }
	 return candidate
}

func majorityElementHelper() {
    testCases := [][]int{
        {3, 2, 3},
        {2, 2, 1, 1, 1, 2, 2},
        {1},
        {6, 5, 5},
    }

    for i, testCase := range testCases {
        result := majorityElement(testCase)
        fmt.Printf("Test Case %d: %d\n", i+1, result)
    }
}
